woocommerce/woocommerce
# Add new E2E test for Shopper Product-related flows

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes # [27](https://github.com/woocommerce/woocommerce-quality/milestone/27) in Woo Quality

Completed the whole milestone but also separated product-related tests into separate test files `product-simple.spec.js`, `product-variable.spec.js` and `product-grouped.spec.js`. Variable tests are now in one test file.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch locally
2. Execute tests:
`pnpm test:e2e-pw ./tests/e2e-pw/tests/shopper/product-grouped.spec.js`
`pnpm test:e2e-pw ./tests/e2e-pw/tests/shopper/product-variable.spec.js`
`pnpm test:e2e-pw ./tests/e2e-pw/tests/shopper/product-simple.spec.js`
3. Make sure they all pass
4. Optionally execute the whole test suite `pnpm test:e2e-pw`



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
